src/view/components/spectrum3ExtensionView.jsx

```diff
@@ -1,78 +1,193 @@
+        validateFormikStatePromise,
+        ...validateNonFormikStatePromises
+      ]);
+      return results.every(result => result);
     }
   });
 
-  // Don't render anything until extension bridge calls init
-  if (!initInfo) {
-    return null;
-  }
+  // If we called getInitialValues as the partial forms are registered it would lead to
+  // a new re-render for every partial form. So after every render of the extension view
+  // we check if there are any uninitialized partial forms and initialized them all
+  // together. Calling formikProps.setValues once only triggers one re-render.
+  useEffect(async () => {
+    if (
+      !initInfo ||
+      registeredPartialFormsRef.current.every(
+        ({ initialized: _initialized }) => _initialized
+      )
+    ) {
+      return;
+    }
+    const allInitialValues = await Promise.all(
+      registeredPartialFormsRef.current
+        .filter(({ initialized: _initialized }) => !_initialized)
+        .map(async ({ getInitialValues, initializedName }) => {
+          const initialValues = await Promise.resolve(
+            getInitialValues({ initInfo })
+          );
+          setValue(initialValues, initializedName, true);
+          return initialValues;
+        })
+    );
+    const newValues = deepAssign(
+      {},
+      formikPropsRef.current.values,
+      ...allInitialValues
+    );
+    // No need to trigger a new validation round here
+    formikPropsRef.current.setValues(newValues, false);
+    if (!initialized) {
+      setInitialized(true);
+    }
+  });
 
-  const context = {
-    registerGetSettings(getSettings) {
-      registeredGetSettingsRef.current.push(getSettings);
-    },
-    deregisterGetSettings(getSettings) {
-      registeredGetSettingsRef.current = registeredGetSettingsRef.current.filter(
-        other => other !== getSettings
-      );
+  const { current: context } = useRef({
+    registerPartialForm(partialForm) {
+      registeredPartialFormsRef.current.push(partialForm);
     },
-    registerValidate(validate) {
-      registeredValidateRef.current.push(validate);
-    },
-    deregisterValidate(validate) {
-      registeredValidateRef.current = registeredValidateRef.current.filter(
-        other => other !== validate
+    dropPartialForm(partialForm) {
+      registeredPartialFormsRef.current = registeredPartialFormsRef.current.filter(
+        other => other !== partialForm
       );
     },
-    initInfo
-  };
+    // TODO: make a resetField and use that instead
+    resetForm(initialValues) {
+      formikPropsRef.current.resetForm({ initialValues });
+    }
+  });
+  context.initInfo = initInfo;
+
+  // Don't render anything until extension bridge calls init
+  if (!initInfo) {
+    return null;
+  }
 
   return (
-    <ExtensionViewContext.Provider value={context}>
-      {render({ initInfo })}
-    </ExtensionViewContext.Provider>
+    <>
+      <ExtensionViewContext.Provider value={context}>
+        <Form>
```

Comment:
I think putting `Form` here might be presumptuous. I think the only real reason we're using `Form` is because it provides a bit of styling on fields. One of the styles (top-margin) is provided through this declaration block:

```
._spectrum-Form_36b9b._spectrum-Form--positionTop_36b9b>* {
    margin-top: var(--spectrum-global-dimension-size-100);
    width: 100%
}
```

Notice that the fields receiving `margin-top` will be the direct children of the form. Considering the `Form` placed here will likely have children that are not the fields (for example, it might be Tabs), I don't think the elements that will be receiving the top margin are the ones we want to be receiving the top margin.

I'm still trying to figure out how much value we're getting out of Form and how to use it most appropriately.